src/wp-includes/blocks.php

```diff
@@ -760,6 +760,86 @@ function get_hooked_blocks( $name ) {
 	return $hooked_blocks;
 }
 
+/**
+ * Returns a function that...
+ *
+ * @since 6.4.0
+ * @access private
+ *
+ * @param WP_Block_Template|array $context A block template, template part, or pattern that the blocks belong to.
+ * @return callable A function that returns the serialized markup for the given block.
+ */
+function make_before_block_visitor( $context ) {
+	return function( &$block, $parent = null, $prev = null ) use ( $context ) {
+		_inject_theme_attribute_in_template_part_block( $block );
+
+		$markup = '';
+
+		if ( $parent && ! $prev ) {
+			// Candidate for first-child insertion.
+			$hooked_blocks_for_parent = get_hooked_blocks( $parent['blockName'] );
```

Comment:
`get_hooked_blocks` is going to be called too many times. I will work on some optimizations tomorrow.